python/GafferUI/AnimationEditor.py

```diff
@@ -868,4 +879,186 @@ def __checkBoxStateForTieMode( self, mode ) :
+				color = curve.colorChangedSignal().connect( Gaffer.WeakMethod( self.__colorChanged ), scoped = False ) )
+
+	def disconnect( self, curve ) :
+		if curve in self.__connections :
+			for connection in self.__connections[ curve ] :
+				connection.disconnect()
+			del self.__connections[ curve ]
+
+	def update( self ) :
+		self.__updateColor()
+
+	def __swatchContextMenu( self, *unused ) :
+
+		menuDefinition = IECore.MenuDefinition()
+		menuDefinition.append(
+			"/Set Default", {
+				"command" : functools.partial( Gaffer.WeakMethod( self.__colorDefault ) ),
+				"active" : bool( self.parent().curveGadget().editablePlugs() )
+			}
+		)
+
+		self.__popupMenu = GafferUI.Menu( menuDefinition )
+		self.__popupMenu.popup( parent = self.__colorSwatch )
+
+		return True
+
+	def __swatchButtonPress( self, widget, event ) :
+
+		# initiate drag from left button press
+		if event.buttons == event.Buttons.Left :
+			return True
+
+		# initiate context menu from right button press
+		if event.buttons == event.Buttons.Right :
+			return self.__swatchContextMenu()
+
+		return False
+
+	def __swatchDragBegin( self, widget, event ) :
+
+		GafferUI.Pointer.setCurrent( "rgba" )
+
+		return self.__colorSwatch.getColor()
+
+	def __swatchDragEnd( self, widget, event ) :
+
+		GafferUI.Pointer.setCurrent( None )
+
+	def __swatchButtonRelease( self, widget, event ) :
+
+		if event.button != event.Buttons.Left :
+			return False
+
+		# check there are selected curves
+		editableCurves = self.parent().curveGadget().editablePlugs()
+		if not editableCurves :
+			return False
+
+		# store initial color of selected curves
+		self.__initialColors.clear()
+		for curve in editableCurves :
+			self.__initialColors[ curve ] = curve.getColor()
+
+		# create color chooser dialog
+		color = self.__colorSwatch.getColor()
+		dialogue = GafferUI.ColorChooserDialogue( color=color, useDisplayTransform=False )
+		dialogue.colorChooser().setInitialColor( color )
+		dialogue.colorChooser().colorChangedSignal().connect( Gaffer.WeakMethod( self.__setColor ), scoped = False )
+		dialogue.cancelButton.clickedSignal().connect( Gaffer.WeakMethod( self.__colorCancelled ), scoped = False )
+
+		# show color chooser dialog as modal
+		dialogue.waitForColor( parentWindow = widget.ancestor( GafferUI.Window ) )
+
+		return True
+
+	def __colorDefault( self ) :
+
+		# ensure seperate entry in undo queue
+		self.__mergeGroupIdColor += 1
+		self.__lastChangedReasonColor = GafferUI.ColorChooser.ColorChangedReason.SetColor
+
+		# restore default color of editable curves
+		editableCurves = self.parent().curveGadget().editablePlugs()
+		if editableCurves :
+			with Gaffer.UndoScope( next( iter( editableCurves ) ).ancestor( Gaffer.ScriptNode ), mergeGroup=str( self.__mergeGroupIdColor ) ) :
+				for curve in editableCurves :
+					with Gaffer.BlockedConnection( self.__connections[ curve ].color ) :
+						curve.setColor( Gaffer.Animation.defaultColor( next( iter( curve[ "out" ].outputs() ) ) ) )
```

Comment:
This is baking the current default into the curve, so that it won't pick up any future changes to the default. If we relied on the Metadata API for everything, we could just call `deregisterValue( curve )` here to remove the value.

Reply:
done